### PR TITLE
Add memory component to Python benchmarking script

### PR DESCRIPTION
**Pull request by FabijanC**
## Usage related changes

- None

## Development related changes

- Documentation updates related to the new functionality
- Documentation improvements after #439

## Checklist:

- [x] Checked out the [contribution guidelines](CONTRIBUTING.md)
- [x] Applied formatting - `./scripts/format.sh`
- [x] No linter errors - `./scripts/clippy_check.sh`
- [x] No unused dependencies - `./scripts/check_unused_deps.sh`
- [x] No spelling errors - `./scripts/check_spelling.sh`
- [x] Performed code self-review
- [x] Rebased to the latest commit of the target branch (or merged it into my branch)
- [x] Updated the docs if needed, including the [TODO section](https://github.com/0xSpaceShard/starknet-devnet-rs/?tab=readme-ov-file#todo-to-reach-feature-parity-with-the-pythonic-devnet)
- [x] Linked the [issues](https://github.com/0xSpaceShard/starknet-devnet-rs/issues) resolvable by this PR - [linking info](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- [x] Updated the tests if needed; all passing - [execution info](https://github.com/0xSpaceShard/starknet-devnet-rs/?tab=readme-ov-file#test-execution)

### CODE REVIEW

File: scripts/benchmark/command_stat_test.py [hunk 1]
```diff
@@ -1,42 +1,46 @@
...
+import psutil
 import requests
 from scipy.stats import ttest_ind
+from scipy.stats import describe
 
 DEVNET_PORT = "5050"
 DEVNET_URL = f"http://localhost:{DEVNET_PORT}"
 REQUEST_TIMEOUT = 2
 
 ORIGINAL_COMMAND: str = ...
```
Comment (mikiw) [inline scripts/benchmark/command_stat_test.py:28]: I'm suggesting adding some default values for `ORIGINAL_COMMAND` and `IMPROVED_COMMAND` so developers can run the script without reading the script itself, maybe:
```
ORIGINAL_COMMAND: str = f"cargo run --release -- --port {DEVNET_PORT}"
IMPROVED_COMMAND: str = f"cargo run --release -- --port {DEVNET_PORT} --lite-mode"
```
?

Later once they know that the script is running correctly they can tweak it for their demands.

**Review by mikiw — CHANGES_REQUESTED** [review]

File: scripts/benchmark/command_stat_test.py [hunk 2]
```diff
@@ -89,45 +93,69 @@ def terminate_and_wait(proc: subprocess.Popen):
...
     Run `command` and run `performance_program` `size` times.
-    Returns a list containing `size` measured times.
+    Returns a tuple of:
+      - a list containing `size` measured times in seconds
+      - a list containing `size` measured memory usages in MB
     """
     total_start_time = time.time()
 
     times = []
+    memories = []
```
[the remark this answers is not in the document]
Comment (FabijanC) [reply scripts/benchmark/command_stat_test.py:106]: Yes, yes, I was also skeptical about that, but decided not to waste to much time on naming while still locally experimenting. But now that it seems it's mergeable, I'll definitely improve the naming.

[the remark this answers is not in the document]
Comment (FabijanC) [reply scripts/benchmark/command_stat_test.py]: Addressed in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/464/commits/16595475df771cdc69754184779a0cd6940bf0b7

File: scripts/benchmark/command_stat_test.py (hunk 1 above)
Comment (FabijanC) [reply scripts/benchmark/command_stat_test.py]: Addressed in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/464/commits/3d17e1ea1cc9caa2058eb6dcee00453d7afaa325

**Pull request merged**